src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerControllerTest.java

```diff
@@ -101,6 +103,83 @@ public void given_initiate_tasks_with_follow_up_overdue_category_then_cancel_tas
         taskToTearDown = taskCreatedAsResultOfTheMultipleDmnRule;
     }
 
+    @Test
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    public void given_initiate_tasks_with_follow_up_overdue_category_then_cancel_all_tasks() {
+
+        // notice this creates two tasks with the follow up category because the initiate dmn table
+        // has multiple rules matching this event and state.
+        String caseIdForTask1 = UUID.randomUUID().toString();
+        String taskIdDmnColumn = "followUpOverdueRespondentEvidence";
+
+        // task1
+        String task1Id = initiateTaskForGivenId(
+            caseIdForTask1,
+            "requestRespondentEvidence",
+            "", "awaitingRespondentEvidence", false,
+            taskIdDmnColumn
+        );
+
+        // task2
+        String task2Id = findTaskForGivenCaseId(
+            caseIdForTask1,
+            "provideRespondentEvidence"
+        );
+
+        // Then cancel all tasks
+        String eventToCancelTask = "removeAppealFromOnline";
+        sendMessage(caseIdForTask1, eventToCancelTask, "", "", false);
+
+        waitSeconds(5);
+        assertTaskDoesNotExist(caseIdForTask1, taskIdDmnColumn);
+        assertTaskDoesNotExist(task2Id, "provideRespondentEvidence");
+
+        assertTaskDeleteReason(task1Id, "deleted");
+        assertTaskDeleteReason(task2Id, "deleted");
+
+        // add tasks to tear down.
+        tearDownMultipleTasks(Arrays.asList(task1Id, task2Id), "deleted");
+    }
+
+    @Test
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    public void given_initiate_tasks_with_different_categories_then_cancel_all_tasks() {
+
+        String caseIdForTask1 = UUID.randomUUID().toString();
+        String task1IdDmnColumn = "reviewTheAppeal";
+
+        // task1 with category Case progression
+        String task1Id = initiateTaskForGivenId(
+            caseIdForTask1,
+            "submitAppeal",
+            "", "appealSubmitted", false,
+            task1IdDmnColumn
+        );
+
+        // task2 with category Time Extension
+        String task2IdDmnColumn = "decideOnTimeExtension";
+        String task2Id = initiateTaskForGivenId(
+            caseIdForTask1,
+            "submitTimeExtension",
+            "", "", false,
+            task2IdDmnColumn
+        );
+
+        // Then cancel all tasks
+        String eventToCancelTask = "removeAppealFromOnline";
+        sendMessage(caseIdForTask1, eventToCancelTask, "", "", false);
+
+        waitSeconds(5);
+        assertTaskDoesNotExist(caseIdForTask1, task1IdDmnColumn);
+        assertTaskDoesNotExist(caseIdForTask1, task2IdDmnColumn);
+
+        assertTaskDeleteReason(task1Id, "deleted");
+        assertTaskDeleteReason(task2Id, "deleted");
+
+        // add tasks to tear down.
+        tearDownMultipleTasks(Arrays.asList(task1Id, task2Id), "deleted");
+    }
+
     @Test
```

Comment:
Can we add tests for warn and cancel scenarios to cover 
1. multiple caseIds+ single category
2. multiple caseIds + multiple categories